Source/Core/Core/PowerPC/SignatureDB/MEGASignatureDB.h

```diff
@@ -0,0 +1,47 @@
+// Copyright 2017 Dolphin Emulator Project
+// Licensed under GPLv2+
+// Refer to the license.txt file included.
+
+#pragma once
+
+#include <sstream>
+#include <string>
+#include <vector>
+
+#include "Common/CommonTypes.h"
+#include "Core/PowerPC/PPCSymbolDB.h"
+
+struct MEGASignatureReference
+{
+  u32 offset;
+  std::string name;
+
+  MEGASignatureReference(u32 _offset, std::string _name) : offset(_offset), name(_name) {}
+};
+
+struct MEGASignature
+{
+  std::vector<u32> code;
+  std::string name;
+  std::vector<MEGASignatureReference> refs;
+};
+
+class MEGASignatureDB
+{
+public:
+  MEGASignatureDB() = default;
```

Comment:
These should be defaulted in the cpp file, otherwise you potentially inline initial set-up and teardown code related to the std::vector in the class, which is usually not desirable.